test/e2e/framework/framework.go

```diff
@@ -262,9 +262,9 @@ func (f *Framework) BeforeEach() {
 		var nodeMode NodesSet
 		switch TestContext.GatherKubeSystemResourceUsageData {
 		case "master":
-			nodeMode = MasterNodes
+			nodeMode = ControlPlaneNodes
```

Comment:
I think it is desirable to deprecate "master" and "masteranddns". Just do it gradually.

That means using `case "master", "controlplane"` here and add a comment that "master" can be removed i Kubernetes 1.23. Add a changelog entry in your PR description.

Also, update:
* https://github.com/kubernetes/kubernetes/blob/ebcb4a2d88c83096e6068aa56e9a5281976e1fec/test/e2e/framework/test_context.go#L130-L133
* https://github.com/kubernetes/kubernetes/blob/ebcb4a2d88c83096e6068aa56e9a5281976e1fec/test/e2e/framework/test_context.go#L290

Finally, please squash your commits.